<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>3.3 How to test for differences between samples | Computational Genomics with R Book Club</title>
  <meta name="description" content="This is the product of the R4DS Online Learning Community’s Computational Genomics with R Book Club." />
  <meta name="generator" content="bookdown 0.30 and GitBook 2.6.7" />

  <meta property="og:title" content="3.3 How to test for differences between samples | Computational Genomics with R Book Club" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is the product of the R4DS Online Learning Community’s Computational Genomics with R Book Club." />
  <meta name="github-repo" content="r4ds/bookclub-comp_genom" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="3.3 How to test for differences between samples | Computational Genomics with R Book Club" />
  
  <meta name="twitter:description" content="This is the product of the R4DS Online Learning Community’s Computational Genomics with R Book Club." />
  

<meta name="author" content="The R4DS Online Learning Community" />


<meta name="date" content="2022-11-22" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="how-to-summarize-collection-of-data-points-the-idea-behind-statistical-distributions.html"/>
<link rel="next" href="relationship-between-variables-linear-models-and-correlation.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Computational Genomics with R Book Club</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a>
<ul>
<li class="chapter" data-level="" data-path="book-club-meetings.html"><a href="book-club-meetings.html"><i class="fa fa-check"></i>Book club meetings</a></li>
<li class="chapter" data-level="" data-path="pace.html"><a href="pace.html"><i class="fa fa-check"></i>Pace</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="introduction-to-genomics.html"><a href="introduction-to-genomics.html"><i class="fa fa-check"></i><b>1</b> Introduction to Genomics</a>
<ul>
<li class="chapter" data-level="1.1" data-path="genes-genomes-and-genomics.html"><a href="genes-genomes-and-genomics.html"><i class="fa fa-check"></i><b>1.1</b> Genes, Genomes, and Genomics</a></li>
<li class="chapter" data-level="1.2" data-path="gene-regulation.html"><a href="gene-regulation.html"><i class="fa fa-check"></i><b>1.2</b> Gene regulation</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="gene-regulation.html"><a href="gene-regulation.html#transcriptional-regulation"><i class="fa fa-check"></i><b>1.2.1</b> Transcriptional regulation</a></li>
<li class="chapter" data-level="1.2.2" data-path="gene-regulation.html"><a href="gene-regulation.html#post-transcriptional-regulation"><i class="fa fa-check"></i><b>1.2.2</b> Post-transcriptional regulation</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="genetic-mutations.html"><a href="genetic-mutations.html"><i class="fa fa-check"></i><b>1.3</b> Genetic mutations</a></li>
<li class="chapter" data-level="1.4" data-path="sequencing.html"><a href="sequencing.html"><i class="fa fa-check"></i><b>1.4</b> Sequencing</a></li>
<li class="chapter" data-level="1.5" data-path="resources.html"><a href="resources.html"><i class="fa fa-check"></i><b>1.5</b> Resources</a></li>
<li class="chapter" data-level="1.6" data-path="meeting-videos.html"><a href="meeting-videos.html"><i class="fa fa-check"></i><b>1.6</b> Meeting Videos</a>
<ul>
<li class="chapter" data-level="1.6.1" data-path="meeting-videos.html"><a href="meeting-videos.html#cohort-1"><i class="fa fa-check"></i><b>1.6.1</b> Cohort 1</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="introduction-to-r-for-genomic-data-analysis.html"><a href="introduction-to-r-for-genomic-data-analysis.html"><i class="fa fa-check"></i><b>2</b> Introduction to R for Genomic Data Analysis</a>
<ul>
<li class="chapter" data-level="2.1" data-path="genomic-data-analysis-a-high-level-view-in-r.html"><a href="genomic-data-analysis-a-high-level-view-in-r.html"><i class="fa fa-check"></i><b>2.1</b> Genomic data analysis (a high level view)– in R!</a></li>
<li class="chapter" data-level="2.2" data-path="getting-started-with-r-and-bioconductor.html"><a href="getting-started-with-r-and-bioconductor.html"><i class="fa fa-check"></i><b>2.2</b> Getting started with R and Bioconductor</a></li>
<li class="chapter" data-level="2.3" data-path="data-stuctures-and-data-types.html"><a href="data-stuctures-and-data-types.html"><i class="fa fa-check"></i><b>2.3</b> Data stuctures and data types</a></li>
<li class="chapter" data-level="2.4" data-path="reading-and-writing-data.html"><a href="reading-and-writing-data.html"><i class="fa fa-check"></i><b>2.4</b> Reading and writing data</a></li>
<li class="chapter" data-level="2.5" data-path="plots-base-r-vs-ggplot.html"><a href="plots-base-r-vs-ggplot.html"><i class="fa fa-check"></i><b>2.5</b> Plots: Base R vs ggplot</a></li>
<li class="chapter" data-level="2.6" data-path="functions-and-loops.html"><a href="functions-and-loops.html"><i class="fa fa-check"></i><b>2.6</b> Functions and Loops</a></li>
<li class="chapter" data-level="2.7" data-path="resources-1.html"><a href="resources-1.html"><i class="fa fa-check"></i><b>2.7</b> Resources</a></li>
<li class="chapter" data-level="2.8" data-path="meeting-videos-1.html"><a href="meeting-videos-1.html"><i class="fa fa-check"></i><b>2.8</b> Meeting Videos</a>
<ul>
<li class="chapter" data-level="2.8.1" data-path="meeting-videos-1.html"><a href="meeting-videos-1.html#cohort-1-1"><i class="fa fa-check"></i><b>2.8.1</b> Cohort 1</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="statistics-for-genomics.html"><a href="statistics-for-genomics.html"><i class="fa fa-check"></i><b>3</b> Statistics for Genomics</a>
<ul>
<li class="chapter" data-level="3.1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="how-to-summarize-collection-of-data-points-the-idea-behind-statistical-distributions.html"><a href="how-to-summarize-collection-of-data-points-the-idea-behind-statistical-distributions.html"><i class="fa fa-check"></i><b>3.2</b> How to summarize collection of data points: The idea behind statistical distributions</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="how-to-summarize-collection-of-data-points-the-idea-behind-statistical-distributions.html"><a href="how-to-summarize-collection-of-data-points-the-idea-behind-statistical-distributions.html#measuring-the-measuring-central-tendency"><i class="fa fa-check"></i><b>3.2.1</b> Measuring the measuring central tendency</a></li>
<li class="chapter" data-level="3.2.2" data-path="how-to-summarize-collection-of-data-points-the-idea-behind-statistical-distributions.html"><a href="how-to-summarize-collection-of-data-points-the-idea-behind-statistical-distributions.html#measurements-of-variation"><i class="fa fa-check"></i><b>3.2.2</b> Measurements of variation</a></li>
<li class="chapter" data-level="3.2.3" data-path="how-to-summarize-collection-of-data-points-the-idea-behind-statistical-distributions.html"><a href="how-to-summarize-collection-of-data-points-the-idea-behind-statistical-distributions.html#statistical-distributions"><i class="fa fa-check"></i><b>3.2.3</b> Statistical distributions</a></li>
<li class="chapter" data-level="3.2.4" data-path="how-to-summarize-collection-of-data-points-the-idea-behind-statistical-distributions.html"><a href="how-to-summarize-collection-of-data-points-the-idea-behind-statistical-distributions.html#confidence-intervals"><i class="fa fa-check"></i><b>3.2.4</b> Confidence intervals</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="how-to-test-for-differences-between-samples.html"><a href="how-to-test-for-differences-between-samples.html"><i class="fa fa-check"></i><b>3.3</b> How to test for differences between samples</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="how-to-test-for-differences-between-samples.html"><a href="how-to-test-for-differences-between-samples.html#randomization"><i class="fa fa-check"></i><b>3.3.1</b> Randomization</a></li>
<li class="chapter" data-level="3.3.2" data-path="how-to-test-for-differences-between-samples.html"><a href="how-to-test-for-differences-between-samples.html#t-test"><i class="fa fa-check"></i><b>3.3.2</b> t-test</a></li>
<li class="chapter" data-level="3.3.3" data-path="how-to-test-for-differences-between-samples.html"><a href="how-to-test-for-differences-between-samples.html#multiple-testing"><i class="fa fa-check"></i><b>3.3.3</b> Multiple testing</a></li>
<li class="chapter" data-level="3.3.4" data-path="how-to-test-for-differences-between-samples.html"><a href="how-to-test-for-differences-between-samples.html#moderated-t-tests"><i class="fa fa-check"></i><b>3.3.4</b> Moderated t-tests</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="relationship-between-variables-linear-models-and-correlation.html"><a href="relationship-between-variables-linear-models-and-correlation.html"><i class="fa fa-check"></i><b>3.4</b> Relationship between variables: Linear models and correlation</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="relationship-between-variables-linear-models-and-correlation.html"><a href="relationship-between-variables-linear-models-and-correlation.html#the-cost-or-loss-function-approach"><i class="fa fa-check"></i><b>3.4.1</b> The cost or loss function approach</a></li>
<li class="chapter" data-level="3.4.2" data-path="relationship-between-variables-linear-models-and-correlation.html"><a href="relationship-between-variables-linear-models-and-correlation.html#the-maximum-likelihood-approach"><i class="fa fa-check"></i><b>3.4.2</b> The “maximum likelihood” approach</a></li>
<li class="chapter" data-level="3.4.3" data-path="relationship-between-variables-linear-models-and-correlation.html"><a href="relationship-between-variables-linear-models-and-correlation.html#linear-algebra-and-closed-form-solution-to-linear-regression"><i class="fa fa-check"></i><b>3.4.3</b> Linear algebra and closed-form solution to linear regression</a></li>
<li class="chapter" data-level="3.4.4" data-path="relationship-between-variables-linear-models-and-correlation.html"><a href="relationship-between-variables-linear-models-and-correlation.html#how-to-estimate-the-error-of-the-coefficients"><i class="fa fa-check"></i><b>3.4.4</b> How to estimate the error of the coefficients</a></li>
<li class="chapter" data-level="3.4.5" data-path="relationship-between-variables-linear-models-and-correlation.html"><a href="relationship-between-variables-linear-models-and-correlation.html#accuracy-of-the-model"><i class="fa fa-check"></i><b>3.4.5</b> Accuracy of the model</a></li>
<li class="chapter" data-level="3.4.6" data-path="relationship-between-variables-linear-models-and-correlation.html"><a href="relationship-between-variables-linear-models-and-correlation.html#regression-with-categorical-variables"><i class="fa fa-check"></i><b>3.4.6</b> Regression with categorical variables</a></li>
<li class="chapter" data-level="3.4.7" data-path="relationship-between-variables-linear-models-and-correlation.html"><a href="relationship-between-variables-linear-models-and-correlation.html#regression-pitfalls"><i class="fa fa-check"></i><b>3.4.7</b> Regression pitfalls</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="meeting-videos-2.html"><a href="meeting-videos-2.html"><i class="fa fa-check"></i><b>3.5</b> Meeting Videos</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="meeting-videos-2.html"><a href="meeting-videos-2.html#cohort-1-2"><i class="fa fa-check"></i><b>3.5.1</b> Cohort 1</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="exploratory-data-analysis-with-unsupervised-machine-learning.html"><a href="exploratory-data-analysis-with-unsupervised-machine-learning.html"><i class="fa fa-check"></i><b>4</b> Exploratory Data Analysis with Unsupervised Machine Learning</a>
<ul>
<li class="chapter" data-level="4.1" data-path="very-helpful-youtube-explanations.html"><a href="very-helpful-youtube-explanations.html"><i class="fa fa-check"></i><b>4.1</b> Very helpful youtube explanations</a></li>
<li class="chapter" data-level="4.2" data-path="chapter-4-exercises.html"><a href="chapter-4-exercises.html"><i class="fa fa-check"></i><b>4.2</b> Chapter 4 Exercises</a></li>
<li class="chapter" data-level="4.3" data-path="meeting-videos-3.html"><a href="meeting-videos-3.html"><i class="fa fa-check"></i><b>4.3</b> Meeting Videos</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="meeting-videos-3.html"><a href="meeting-videos-3.html#cohort-1-3"><i class="fa fa-check"></i><b>4.3.1</b> Cohort 1</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="predictive-modeling-with-supervised-machine-learning.html"><a href="predictive-modeling-with-supervised-machine-learning.html"><i class="fa fa-check"></i><b>5</b> Predictive Modeling with Supervised Machine Learning</a>
<ul>
<li class="chapter" data-level="5.1" data-path="and-5.html"><a href="and-5.html"><i class="fa fa-check"></i><b>5.1</b> 5.1 and 5.2</a></li>
<li class="chapter" data-level="5.2" data-path="data-pre-processing.html"><a href="data-pre-processing.html"><i class="fa fa-check"></i><b>5.2</b> 5.4 Data pre-processing</a></li>
<li class="chapter" data-level="5.3" data-path="split-the-data.html"><a href="split-the-data.html"><i class="fa fa-check"></i><b>5.3</b> 5.5 Split the data</a></li>
<li class="chapter" data-level="5.4" data-path="some-unsupervised-learning-on-training-data-to-initially-cluster-data.html"><a href="some-unsupervised-learning-on-training-data-to-initially-cluster-data.html"><i class="fa fa-check"></i><b>5.4</b> 5.6 Some unsupervised learning on training data to initially cluster data</a></li>
<li class="chapter" data-level="5.5" data-path="section.html"><a href="section.html"><i class="fa fa-check"></i><b>5.5</b> 5.7</a></li>
<li class="chapter" data-level="5.6" data-path="section-1.html"><a href="section-1.html"><i class="fa fa-check"></i><b>5.6</b> 5.8</a></li>
<li class="chapter" data-level="5.7" data-path="meeting-videos-4.html"><a href="meeting-videos-4.html"><i class="fa fa-check"></i><b>5.7</b> Meeting Videos</a>
<ul>
<li class="chapter" data-level="5.7.1" data-path="meeting-videos-4.html"><a href="meeting-videos-4.html#cohort-1-4"><i class="fa fa-check"></i><b>5.7.1</b> Cohort 1</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="operations-on-genomic-intervals-and-genome-arithmetic.html"><a href="operations-on-genomic-intervals-and-genome-arithmetic.html"><i class="fa fa-check"></i><b>6</b> Operations on Genomic Intervals and Genome Arithmetic</a>
<ul>
<li class="chapter" data-level="6.1" data-path="slide-1.html"><a href="slide-1.html"><i class="fa fa-check"></i><b>6.1</b> SLIDE 1</a></li>
<li class="chapter" data-level="6.2" data-path="meeting-videos-5.html"><a href="meeting-videos-5.html"><i class="fa fa-check"></i><b>6.2</b> Meeting Videos</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="meeting-videos-5.html"><a href="meeting-videos-5.html#cohort-1-5"><i class="fa fa-check"></i><b>6.2.1</b> Cohort 1</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="quality-check-processing-and-alignment-of-high-throughput-sequencing-reads.html"><a href="quality-check-processing-and-alignment-of-high-throughput-sequencing-reads.html"><i class="fa fa-check"></i><b>7</b> Quality Check, Processing and Alignment of High-throughput Sequencing Reads</a>
<ul>
<li class="chapter" data-level="7.1" data-path="slide-1-1.html"><a href="slide-1-1.html"><i class="fa fa-check"></i><b>7.1</b> SLIDE 1</a></li>
<li class="chapter" data-level="7.2" data-path="meeting-videos-6.html"><a href="meeting-videos-6.html"><i class="fa fa-check"></i><b>7.2</b> Meeting Videos</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="meeting-videos-6.html"><a href="meeting-videos-6.html#cohort-1-6"><i class="fa fa-check"></i><b>7.2.1</b> Cohort 1</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="rna-seq-analysis.html"><a href="rna-seq-analysis.html"><i class="fa fa-check"></i><b>8</b> RNA-seq Analysis</a>
<ul>
<li class="chapter" data-level="8.1" data-path="slide-1-2.html"><a href="slide-1-2.html"><i class="fa fa-check"></i><b>8.1</b> SLIDE 1</a></li>
<li class="chapter" data-level="8.2" data-path="meeting-videos-7.html"><a href="meeting-videos-7.html"><i class="fa fa-check"></i><b>8.2</b> Meeting Videos</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="meeting-videos-7.html"><a href="meeting-videos-7.html#cohort-1-7"><i class="fa fa-check"></i><b>8.2.1</b> Cohort 1</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="chip-seq-analysis.html"><a href="chip-seq-analysis.html"><i class="fa fa-check"></i><b>9</b> ChIP-seq analysis</a>
<ul>
<li class="chapter" data-level="9.1" data-path="slide-1-3.html"><a href="slide-1-3.html"><i class="fa fa-check"></i><b>9.1</b> SLIDE 1</a></li>
<li class="chapter" data-level="9.2" data-path="meeting-videos-8.html"><a href="meeting-videos-8.html"><i class="fa fa-check"></i><b>9.2</b> Meeting Videos</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="meeting-videos-8.html"><a href="meeting-videos-8.html#cohort-1-8"><i class="fa fa-check"></i><b>9.2.1</b> Cohort 1</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="dna-methylation-analysis-using-bisulfite-sequencing-data.html"><a href="dna-methylation-analysis-using-bisulfite-sequencing-data.html"><i class="fa fa-check"></i><b>10</b> DNA methylation analysis using bisulfite sequencing data</a>
<ul>
<li class="chapter" data-level="10.1" data-path="slide-1-4.html"><a href="slide-1-4.html"><i class="fa fa-check"></i><b>10.1</b> SLIDE 1</a></li>
<li class="chapter" data-level="10.2" data-path="meeting-videos-9.html"><a href="meeting-videos-9.html"><i class="fa fa-check"></i><b>10.2</b> Meeting Videos</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="meeting-videos-9.html"><a href="meeting-videos-9.html#cohort-1-9"><i class="fa fa-check"></i><b>10.2.1</b> Cohort 1</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="multi-omics-analysis.html"><a href="multi-omics-analysis.html"><i class="fa fa-check"></i><b>11</b> Multi-omics Analysis</a>
<ul>
<li class="chapter" data-level="11.1" data-path="slide-1-5.html"><a href="slide-1-5.html"><i class="fa fa-check"></i><b>11.1</b> SLIDE 1</a></li>
<li class="chapter" data-level="11.2" data-path="meeting-videos-10.html"><a href="meeting-videos-10.html"><i class="fa fa-check"></i><b>11.2</b> Meeting Videos</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="meeting-videos-10.html"><a href="meeting-videos-10.html#cohort-1-10"><i class="fa fa-check"></i><b>11.2.1</b> Cohort 1</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Computational Genomics with R Book Club</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="how-to-test-for-differences-between-samples" class="section level2 hasAnchor" number="3.3">
<h2><span class="header-section-number">3.3</span> How to test for differences between samples<a href="how-to-test-for-differences-between-samples.html#how-to-test-for-differences-between-samples" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Variability in our measurements
healthy samples are different from disease samples in some measurable feature (blood count, gene expression, methylation of certain loci).</p>
<ol style="list-style-type: decimal">
<li>subtract the means of two samples</li>
<li>hypothesis testing</li>
</ol>
<blockquote>
<p>“we can compare the real difference and measure how unlikely it is to get such a value under the expectation of the null hypothesis”</p>
</blockquote>
<div id="randomization" class="section level3 hasAnchor" number="3.3.1">
<h3><span class="header-section-number">3.3.1</span> Randomization<a href="how-to-test-for-differences-between-samples.html#randomization" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="how-to-test-for-differences-between-samples.html#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">100</span>)</span>
<span id="cb29-2"><a href="how-to-test-for-differences-between-samples.html#cb29-2" aria-hidden="true" tabindex="-1"></a>gene1<span class="ot">=</span><span class="fu">rnorm</span>(<span class="dv">30</span>,<span class="at">mean=</span><span class="dv">4</span>,<span class="at">sd=</span><span class="dv">2</span>)</span>
<span id="cb29-3"><a href="how-to-test-for-differences-between-samples.html#cb29-3" aria-hidden="true" tabindex="-1"></a>gene2<span class="ot">=</span><span class="fu">rnorm</span>(<span class="dv">30</span>,<span class="at">mean=</span><span class="dv">2</span>,<span class="at">sd=</span><span class="dv">2</span>)</span>
<span id="cb29-4"><a href="how-to-test-for-differences-between-samples.html#cb29-4" aria-hidden="true" tabindex="-1"></a>org.diff<span class="ot">=</span><span class="fu">mean</span>(gene1)<span class="sc">-</span><span class="fu">mean</span>(gene2)</span>
<span id="cb29-5"><a href="how-to-test-for-differences-between-samples.html#cb29-5" aria-hidden="true" tabindex="-1"></a>gene.df<span class="ot">=</span><span class="fu">data.frame</span>(<span class="at">exp=</span><span class="fu">c</span>(gene1,gene2),</span>
<span id="cb29-6"><a href="how-to-test-for-differences-between-samples.html#cb29-6" aria-hidden="true" tabindex="-1"></a>                  <span class="at">group=</span><span class="fu">c</span>( <span class="fu">rep</span>(<span class="st">&quot;test&quot;</span>,<span class="dv">30</span>),<span class="fu">rep</span>(<span class="st">&quot;control&quot;</span>,<span class="dv">30</span>) ) )</span>
<span id="cb29-7"><a href="how-to-test-for-differences-between-samples.html#cb29-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-8"><a href="how-to-test-for-differences-between-samples.html#cb29-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-9"><a href="how-to-test-for-differences-between-samples.html#cb29-9" aria-hidden="true" tabindex="-1"></a>exp.null <span class="ot">&lt;-</span> <span class="fu">do</span>(<span class="dv">1000</span>) <span class="sc">*</span> <span class="fu">diff</span>(mosaic<span class="sc">::</span><span class="fu">mean</span>(exp <span class="sc">~</span> <span class="fu">shuffle</span>(group), <span class="at">data=</span>gene.df))</span>
<span id="cb29-10"><a href="how-to-test-for-differences-between-samples.html#cb29-10" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(exp.null[,<span class="dv">1</span>],<span class="at">xlab=</span><span class="st">&quot;null distribution | no difference in samples&quot;</span>,</span>
<span id="cb29-11"><a href="how-to-test-for-differences-between-samples.html#cb29-11" aria-hidden="true" tabindex="-1"></a>     <span class="at">main=</span><span class="fu">expression</span>(<span class="fu">paste</span>(H[<span class="dv">0</span>],<span class="st">&quot; :no difference in means&quot;</span>) ),</span>
<span id="cb29-12"><a href="how-to-test-for-differences-between-samples.html#cb29-12" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">2</span>,<span class="dv">2</span>),<span class="at">col=</span><span class="st">&quot;cornflowerblue&quot;</span>,<span class="at">border=</span><span class="st">&quot;white&quot;</span>)</span>
<span id="cb29-13"><a href="how-to-test-for-differences-between-samples.html#cb29-13" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">v=</span><span class="fu">quantile</span>(exp.null[,<span class="dv">1</span>],<span class="fl">0.95</span>),<span class="at">col=</span><span class="st">&quot;red&quot;</span> )</span>
<span id="cb29-14"><a href="how-to-test-for-differences-between-samples.html#cb29-14" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">v=</span>org.diff,<span class="at">col=</span><span class="st">&quot;blue&quot;</span> )</span>
<span id="cb29-15"><a href="how-to-test-for-differences-between-samples.html#cb29-15" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(<span class="at">x=</span><span class="fu">quantile</span>(exp.null[,<span class="dv">1</span>],<span class="fl">0.95</span>),<span class="at">y=</span><span class="dv">200</span>,<span class="st">&quot;0.05&quot;</span>,<span class="at">adj=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">0</span>),<span class="at">col=</span><span class="st">&quot;red&quot;</span>)</span>
<span id="cb29-16"><a href="how-to-test-for-differences-between-samples.html#cb29-16" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(<span class="at">x=</span>org.diff,<span class="at">y=</span><span class="dv">200</span>,<span class="st">&quot;org. diff.&quot;</span>,<span class="at">adj=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">0</span>),<span class="at">col=</span><span class="st">&quot;blue&quot;</span>)</span></code></pre></div>
<p><img src="bookclub-comp_genom_files/figure-html/03-2-1.png" width="672" /></p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="how-to-test-for-differences-between-samples.html#cb30-1" aria-hidden="true" tabindex="-1"></a>p.val<span class="ot">=</span><span class="fu">sum</span>(exp.null[,<span class="dv">1</span>]<span class="sc">&gt;</span>org.diff)<span class="sc">/</span><span class="fu">length</span>(exp.null[,<span class="dv">1</span>])</span>
<span id="cb30-2"><a href="how-to-test-for-differences-between-samples.html#cb30-2" aria-hidden="true" tabindex="-1"></a>p.val</span></code></pre></div>
<pre><code>## [1] 0.001</code></pre>
</div>
<div id="t-test" class="section level3 hasAnchor" number="3.3.2">
<h3><span class="header-section-number">3.3.2</span> t-test<a href="how-to-test-for-differences-between-samples.html#t-test" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="how-to-test-for-differences-between-samples.html#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Welch&#39;s t-test</span></span>
<span id="cb32-2"><a href="how-to-test-for-differences-between-samples.html#cb32-2" aria-hidden="true" tabindex="-1"></a>stats<span class="sc">::</span><span class="fu">t.test</span>(gene1,gene2)</span></code></pre></div>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  gene1 and gene2
## t = 3.7653, df = 47.552, p-value = 0.0004575
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  0.872397 2.872761
## sample estimates:
## mean of x mean of y 
##  4.057728  2.185149</code></pre>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="how-to-test-for-differences-between-samples.html#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="co"># t-test with equal variance assumption</span></span>
<span id="cb34-2"><a href="how-to-test-for-differences-between-samples.html#cb34-2" aria-hidden="true" tabindex="-1"></a>stats<span class="sc">::</span><span class="fu">t.test</span>(gene1,gene2,<span class="at">var.equal=</span><span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## 
##  Two Sample t-test
## 
## data:  gene1 and gene2
## t = 3.7653, df = 58, p-value = 0.0003905
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  0.8770753 2.8680832
## sample estimates:
## mean of x mean of y 
##  4.057728  2.185149</code></pre>
</div>
<div id="multiple-testing" class="section level3 hasAnchor" number="3.3.3">
<h3><span class="header-section-number">3.3.3</span> Multiple testing<a href="how-to-test-for-differences-between-samples.html#multiple-testing" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="how-to-test-for-differences-between-samples.html#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="co"># BiocManager::install(&quot;qvalue&quot;)</span></span>
<span id="cb36-2"><a href="how-to-test-for-differences-between-samples.html#cb36-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(qvalue)</span>
<span id="cb36-3"><a href="how-to-test-for-differences-between-samples.html#cb36-3" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(hedenfalk)</span>
<span id="cb36-4"><a href="how-to-test-for-differences-between-samples.html#cb36-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-5"><a href="how-to-test-for-differences-between-samples.html#cb36-5" aria-hidden="true" tabindex="-1"></a>qvalues <span class="ot">&lt;-</span> <span class="fu">qvalue</span>(hedenfalk<span class="sc">$</span>p)<span class="sc">$</span>q</span>
<span id="cb36-6"><a href="how-to-test-for-differences-between-samples.html#cb36-6" aria-hidden="true" tabindex="-1"></a>bonf.pval<span class="ot">=</span><span class="fu">p.adjust</span>(hedenfalk<span class="sc">$</span>p,<span class="at">method =</span><span class="st">&quot;bonferroni&quot;</span>)</span>
<span id="cb36-7"><a href="how-to-test-for-differences-between-samples.html#cb36-7" aria-hidden="true" tabindex="-1"></a>fdr.adj.pval<span class="ot">=</span><span class="fu">p.adjust</span>(hedenfalk<span class="sc">$</span>p,<span class="at">method =</span><span class="st">&quot;fdr&quot;</span>)</span>
<span id="cb36-8"><a href="how-to-test-for-differences-between-samples.html#cb36-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-9"><a href="how-to-test-for-differences-between-samples.html#cb36-9" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(hedenfalk<span class="sc">$</span>p,qvalues,<span class="at">pch=</span><span class="dv">19</span>,<span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>),</span>
<span id="cb36-10"><a href="how-to-test-for-differences-between-samples.html#cb36-10" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab=</span><span class="st">&quot;raw P-values&quot;</span>,<span class="at">ylab=</span><span class="st">&quot;adjusted P-values&quot;</span>)</span>
<span id="cb36-11"><a href="how-to-test-for-differences-between-samples.html#cb36-11" aria-hidden="true" tabindex="-1"></a><span class="fu">points</span>(hedenfalk<span class="sc">$</span>p,bonf.pval,<span class="at">pch=</span><span class="dv">19</span>,<span class="at">col=</span><span class="st">&quot;red&quot;</span>)</span>
<span id="cb36-12"><a href="how-to-test-for-differences-between-samples.html#cb36-12" aria-hidden="true" tabindex="-1"></a><span class="fu">points</span>(hedenfalk<span class="sc">$</span>p,fdr.adj.pval,<span class="at">pch=</span><span class="dv">19</span>,<span class="at">col=</span><span class="st">&quot;blue&quot;</span>)</span>
<span id="cb36-13"><a href="how-to-test-for-differences-between-samples.html#cb36-13" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">&quot;bottomright&quot;</span>,<span class="at">legend=</span><span class="fu">c</span>(<span class="st">&quot;q-value&quot;</span>,<span class="st">&quot;FDR (BH)&quot;</span>,<span class="st">&quot;Bonferroni&quot;</span>),</span>
<span id="cb36-14"><a href="how-to-test-for-differences-between-samples.html#cb36-14" aria-hidden="true" tabindex="-1"></a>       <span class="at">fill=</span><span class="fu">c</span>(<span class="st">&quot;black&quot;</span>,<span class="st">&quot;blue&quot;</span>,<span class="st">&quot;red&quot;</span>))</span></code></pre></div>
<p><img src="bookclub-comp_genom_files/figure-html/03-6-1.png" width="672" /></p>
</div>
<div id="moderated-t-tests" class="section level3 hasAnchor" number="3.3.4">
<h3><span class="header-section-number">3.3.4</span> Moderated t-tests<a href="how-to-test-for-differences-between-samples.html#moderated-t-tests" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For example, if you have many variances calculated for thousands of genes across samples, you can force individual variance estimates to shrink toward the mean or the median of the distribution of variances.</p>
<p><strong>How much the values are shrunk toward a common value depends on the exact method used. These tests in general are called moderated t-tests or shrinkage t-tests.</strong></p>
<ol style="list-style-type: decimal">
<li>One approach popularized by Limma software is to use so-called “Empirical Bayes methods”. The main formulation in these methods is</li>
</ol>
<p><span class="math display">\[\hat{V_g}=aV_0+bV_g\]</span></p>
<p><span class="math display">\[V_0=\text{background variability (the prior)}\]</span>
<span class="math display">\[V_g=\text{is the individual variability}\]</span></p>
<p><span class="math display">\[\hat{V_g}=\text{“shrunk” version of the variability.}\]</span></p>
<p>is the background variability and is the individual variability. Then, these methods estimate and in various ways to come up with a “shrunk” version of the variability. Bayesian inference can make use of prior knowledge to make inference about properties of the data. In a Bayesian viewpoint, the prior knowledge, in this case variability of other genes, can be used to calculate the variability of an individual gene. In our case, would be the prior knowledge we have on the variability of the genes and we use that knowledge to influence our estimate for the individual genes.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="how-to-test-for-differences-between-samples.html#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">100</span>)</span>
<span id="cb37-2"><a href="how-to-test-for-differences-between-samples.html#cb37-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-3"><a href="how-to-test-for-differences-between-samples.html#cb37-3" aria-hidden="true" tabindex="-1"></a><span class="co">#sample data matrix from normal distribution</span></span>
<span id="cb37-4"><a href="how-to-test-for-differences-between-samples.html#cb37-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-5"><a href="how-to-test-for-differences-between-samples.html#cb37-5" aria-hidden="true" tabindex="-1"></a>gset<span class="ot">=</span><span class="fu">rnorm</span>(<span class="dv">3000</span>,<span class="at">mean=</span><span class="dv">200</span>,<span class="at">sd=</span><span class="dv">70</span>)</span>
<span id="cb37-6"><a href="how-to-test-for-differences-between-samples.html#cb37-6" aria-hidden="true" tabindex="-1"></a>data<span class="ot">=</span><span class="fu">matrix</span>(gset,<span class="at">ncol=</span><span class="dv">6</span>)</span>
<span id="cb37-7"><a href="how-to-test-for-differences-between-samples.html#cb37-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-8"><a href="how-to-test-for-differences-between-samples.html#cb37-8" aria-hidden="true" tabindex="-1"></a><span class="co"># set groups</span></span>
<span id="cb37-9"><a href="how-to-test-for-differences-between-samples.html#cb37-9" aria-hidden="true" tabindex="-1"></a>group1<span class="ot">=</span><span class="dv">1</span><span class="sc">:</span><span class="dv">3</span></span>
<span id="cb37-10"><a href="how-to-test-for-differences-between-samples.html#cb37-10" aria-hidden="true" tabindex="-1"></a>group2<span class="ot">=</span><span class="dv">4</span><span class="sc">:</span><span class="dv">6</span></span>
<span id="cb37-11"><a href="how-to-test-for-differences-between-samples.html#cb37-11" aria-hidden="true" tabindex="-1"></a>n1<span class="ot">=</span><span class="dv">3</span></span>
<span id="cb37-12"><a href="how-to-test-for-differences-between-samples.html#cb37-12" aria-hidden="true" tabindex="-1"></a>n2<span class="ot">=</span><span class="dv">3</span></span>
<span id="cb37-13"><a href="how-to-test-for-differences-between-samples.html#cb37-13" aria-hidden="true" tabindex="-1"></a>dx<span class="ot">=</span><span class="fu">rowMeans</span>(data[,group1])<span class="sc">-</span><span class="fu">rowMeans</span>(data[,group2])</span>
<span id="cb37-14"><a href="how-to-test-for-differences-between-samples.html#cb37-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb37-15"><a href="how-to-test-for-differences-between-samples.html#cb37-15" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(matrixStats)</span>
<span id="cb37-16"><a href="how-to-test-for-differences-between-samples.html#cb37-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-17"><a href="how-to-test-for-differences-between-samples.html#cb37-17" aria-hidden="true" tabindex="-1"></a><span class="co"># get the estimate of pooled variance </span></span>
<span id="cb37-18"><a href="how-to-test-for-differences-between-samples.html#cb37-18" aria-hidden="true" tabindex="-1"></a>stderr <span class="ot">=</span> <span class="fu">sqrt</span>( (<span class="fu">rowVars</span>(data[,group1])<span class="sc">*</span>(n1<span class="dv">-1</span>) <span class="sc">+</span> </span>
<span id="cb37-19"><a href="how-to-test-for-differences-between-samples.html#cb37-19" aria-hidden="true" tabindex="-1"></a>       <span class="fu">rowVars</span>(data[,group2])<span class="sc">*</span>(n2<span class="dv">-1</span>)) <span class="sc">/</span> (n1<span class="sc">+</span>n2<span class="dv">-2</span>) <span class="sc">*</span> ( <span class="dv">1</span><span class="sc">/</span>n1 <span class="sc">+</span> <span class="dv">1</span><span class="sc">/</span>n2 ))</span>
<span id="cb37-20"><a href="how-to-test-for-differences-between-samples.html#cb37-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-21"><a href="how-to-test-for-differences-between-samples.html#cb37-21" aria-hidden="true" tabindex="-1"></a><span class="co"># do the shrinking towards median</span></span>
<span id="cb37-22"><a href="how-to-test-for-differences-between-samples.html#cb37-22" aria-hidden="true" tabindex="-1"></a>mod.stderr <span class="ot">=</span> (stderr <span class="sc">+</span> <span class="fu">median</span>(stderr)) <span class="sc">/</span> <span class="dv">2</span> <span class="co"># moderation in variation</span></span>
<span id="cb37-23"><a href="how-to-test-for-differences-between-samples.html#cb37-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-24"><a href="how-to-test-for-differences-between-samples.html#cb37-24" aria-hidden="true" tabindex="-1"></a><span class="co"># estimate t statistic with moderated variance</span></span>
<span id="cb37-25"><a href="how-to-test-for-differences-between-samples.html#cb37-25" aria-hidden="true" tabindex="-1"></a>t.mod <span class="ot">&lt;-</span> dx <span class="sc">/</span> mod.stderr</span>
<span id="cb37-26"><a href="how-to-test-for-differences-between-samples.html#cb37-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-27"><a href="how-to-test-for-differences-between-samples.html#cb37-27" aria-hidden="true" tabindex="-1"></a><span class="co"># calculate P-value of rejecting null </span></span>
<span id="cb37-28"><a href="how-to-test-for-differences-between-samples.html#cb37-28" aria-hidden="true" tabindex="-1"></a>p.mod <span class="ot">=</span> <span class="dv">2</span><span class="sc">*</span><span class="fu">pt</span>( <span class="sc">-</span><span class="fu">abs</span>(t.mod), n1<span class="sc">+</span>n2<span class="dv">-2</span> )</span>
<span id="cb37-29"><a href="how-to-test-for-differences-between-samples.html#cb37-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-30"><a href="how-to-test-for-differences-between-samples.html#cb37-30" aria-hidden="true" tabindex="-1"></a><span class="co"># estimate t statistic without moderated variance</span></span>
<span id="cb37-31"><a href="how-to-test-for-differences-between-samples.html#cb37-31" aria-hidden="true" tabindex="-1"></a>t <span class="ot">=</span> dx <span class="sc">/</span> stderr</span>
<span id="cb37-32"><a href="how-to-test-for-differences-between-samples.html#cb37-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-33"><a href="how-to-test-for-differences-between-samples.html#cb37-33" aria-hidden="true" tabindex="-1"></a><span class="co"># calculate P-value of rejecting null </span></span>
<span id="cb37-34"><a href="how-to-test-for-differences-between-samples.html#cb37-34" aria-hidden="true" tabindex="-1"></a>p <span class="ot">=</span> <span class="dv">2</span><span class="sc">*</span><span class="fu">pt</span>( <span class="sc">-</span><span class="fu">abs</span>(t), n1<span class="sc">+</span>n2<span class="dv">-2</span> )</span>
<span id="cb37-35"><a href="how-to-test-for-differences-between-samples.html#cb37-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-36"><a href="how-to-test-for-differences-between-samples.html#cb37-36" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb37-37"><a href="how-to-test-for-differences-between-samples.html#cb37-37" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(p,<span class="at">col=</span><span class="st">&quot;cornflowerblue&quot;</span>,<span class="at">border=</span><span class="st">&quot;white&quot;</span>,<span class="at">main=</span><span class="st">&quot;&quot;</span>,<span class="at">xlab=</span><span class="st">&quot;P-values t-test&quot;</span>)</span>
<span id="cb37-38"><a href="how-to-test-for-differences-between-samples.html#cb37-38" aria-hidden="true" tabindex="-1"></a><span class="fu">mtext</span>(<span class="fu">paste</span>(<span class="st">&quot;signifcant tests:&quot;</span>,<span class="fu">sum</span>(p<span class="sc">&lt;</span><span class="fl">0.05</span>))  )</span>
<span id="cb37-39"><a href="how-to-test-for-differences-between-samples.html#cb37-39" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(p.mod,<span class="at">col=</span><span class="st">&quot;cornflowerblue&quot;</span>,<span class="at">border=</span><span class="st">&quot;white&quot;</span>,<span class="at">main=</span><span class="st">&quot;&quot;</span>,</span>
<span id="cb37-40"><a href="how-to-test-for-differences-between-samples.html#cb37-40" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab=</span><span class="st">&quot;P-values mod. t-test&quot;</span>)</span>
<span id="cb37-41"><a href="how-to-test-for-differences-between-samples.html#cb37-41" aria-hidden="true" tabindex="-1"></a><span class="fu">mtext</span>(<span class="fu">paste</span>(<span class="st">&quot;signifcant tests:&quot;</span>,<span class="fu">sum</span>(p.mod<span class="sc">&lt;</span><span class="fl">0.05</span>))  )</span></code></pre></div>
<p><img src="bookclub-comp_genom_files/figure-html/03-7-1.png" width="672" /></p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="how-to-summarize-collection-of-data-points-the-idea-behind-statistical-distributions.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="relationship-between-variables-linear-models-and-correlation.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/r4ds/bookclub-comp_genom/edit/main/03.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
