<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>2.4 Reading and writing data | Computational Genomics with R Book Club</title>
  <meta name="description" content="This is the product of the R4DS Online Learning Community’s Computational Genomics with R Book Club." />
  <meta name="generator" content="bookdown 0.30 and GitBook 2.6.7" />

  <meta property="og:title" content="2.4 Reading and writing data | Computational Genomics with R Book Club" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is the product of the R4DS Online Learning Community’s Computational Genomics with R Book Club." />
  <meta name="github-repo" content="r4ds/bookclub-comp_genom" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="2.4 Reading and writing data | Computational Genomics with R Book Club" />
  
  <meta name="twitter:description" content="This is the product of the R4DS Online Learning Community’s Computational Genomics with R Book Club." />
  

<meta name="author" content="The R4DS Online Learning Community" />


<meta name="date" content="2022-11-22" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="data-stuctures-and-data-types.html"/>
<link rel="next" href="plots-base-r-vs-ggplot.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Computational Genomics with R Book Club</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a>
<ul>
<li class="chapter" data-level="" data-path="book-club-meetings.html"><a href="book-club-meetings.html"><i class="fa fa-check"></i>Book club meetings</a></li>
<li class="chapter" data-level="" data-path="pace.html"><a href="pace.html"><i class="fa fa-check"></i>Pace</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="introduction-to-genomics.html"><a href="introduction-to-genomics.html"><i class="fa fa-check"></i><b>1</b> Introduction to Genomics</a>
<ul>
<li class="chapter" data-level="1.1" data-path="genes-genomes-and-genomics.html"><a href="genes-genomes-and-genomics.html"><i class="fa fa-check"></i><b>1.1</b> Genes, Genomes, and Genomics</a></li>
<li class="chapter" data-level="1.2" data-path="gene-regulation.html"><a href="gene-regulation.html"><i class="fa fa-check"></i><b>1.2</b> Gene regulation</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="gene-regulation.html"><a href="gene-regulation.html#transcriptional-regulation"><i class="fa fa-check"></i><b>1.2.1</b> Transcriptional regulation</a></li>
<li class="chapter" data-level="1.2.2" data-path="gene-regulation.html"><a href="gene-regulation.html#post-transcriptional-regulation"><i class="fa fa-check"></i><b>1.2.2</b> Post-transcriptional regulation</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="genetic-mutations.html"><a href="genetic-mutations.html"><i class="fa fa-check"></i><b>1.3</b> Genetic mutations</a></li>
<li class="chapter" data-level="1.4" data-path="sequencing.html"><a href="sequencing.html"><i class="fa fa-check"></i><b>1.4</b> Sequencing</a></li>
<li class="chapter" data-level="1.5" data-path="resources.html"><a href="resources.html"><i class="fa fa-check"></i><b>1.5</b> Resources</a></li>
<li class="chapter" data-level="1.6" data-path="meeting-videos.html"><a href="meeting-videos.html"><i class="fa fa-check"></i><b>1.6</b> Meeting Videos</a>
<ul>
<li class="chapter" data-level="1.6.1" data-path="meeting-videos.html"><a href="meeting-videos.html#cohort-1"><i class="fa fa-check"></i><b>1.6.1</b> Cohort 1</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="introduction-to-r-for-genomic-data-analysis.html"><a href="introduction-to-r-for-genomic-data-analysis.html"><i class="fa fa-check"></i><b>2</b> Introduction to R for Genomic Data Analysis</a>
<ul>
<li class="chapter" data-level="2.1" data-path="genomic-data-analysis-a-high-level-view-in-r.html"><a href="genomic-data-analysis-a-high-level-view-in-r.html"><i class="fa fa-check"></i><b>2.1</b> Genomic data analysis (a high level view)– in R!</a></li>
<li class="chapter" data-level="2.2" data-path="getting-started-with-r-and-bioconductor.html"><a href="getting-started-with-r-and-bioconductor.html"><i class="fa fa-check"></i><b>2.2</b> Getting started with R and Bioconductor</a></li>
<li class="chapter" data-level="2.3" data-path="data-stuctures-and-data-types.html"><a href="data-stuctures-and-data-types.html"><i class="fa fa-check"></i><b>2.3</b> Data stuctures and data types</a></li>
<li class="chapter" data-level="2.4" data-path="reading-and-writing-data.html"><a href="reading-and-writing-data.html"><i class="fa fa-check"></i><b>2.4</b> Reading and writing data</a></li>
<li class="chapter" data-level="2.5" data-path="plots-base-r-vs-ggplot.html"><a href="plots-base-r-vs-ggplot.html"><i class="fa fa-check"></i><b>2.5</b> Plots: Base R vs ggplot</a></li>
<li class="chapter" data-level="2.6" data-path="functions-and-loops.html"><a href="functions-and-loops.html"><i class="fa fa-check"></i><b>2.6</b> Functions and Loops</a></li>
<li class="chapter" data-level="2.7" data-path="resources-1.html"><a href="resources-1.html"><i class="fa fa-check"></i><b>2.7</b> Resources</a></li>
<li class="chapter" data-level="2.8" data-path="meeting-videos-1.html"><a href="meeting-videos-1.html"><i class="fa fa-check"></i><b>2.8</b> Meeting Videos</a>
<ul>
<li class="chapter" data-level="2.8.1" data-path="meeting-videos-1.html"><a href="meeting-videos-1.html#cohort-1-1"><i class="fa fa-check"></i><b>2.8.1</b> Cohort 1</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="statistics-for-genomics.html"><a href="statistics-for-genomics.html"><i class="fa fa-check"></i><b>3</b> Statistics for Genomics</a>
<ul>
<li class="chapter" data-level="3.1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="how-to-summarize-collection-of-data-points-the-idea-behind-statistical-distributions.html"><a href="how-to-summarize-collection-of-data-points-the-idea-behind-statistical-distributions.html"><i class="fa fa-check"></i><b>3.2</b> How to summarize collection of data points: The idea behind statistical distributions</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="how-to-summarize-collection-of-data-points-the-idea-behind-statistical-distributions.html"><a href="how-to-summarize-collection-of-data-points-the-idea-behind-statistical-distributions.html#measuring-the-measuring-central-tendency"><i class="fa fa-check"></i><b>3.2.1</b> Measuring the measuring central tendency</a></li>
<li class="chapter" data-level="3.2.2" data-path="how-to-summarize-collection-of-data-points-the-idea-behind-statistical-distributions.html"><a href="how-to-summarize-collection-of-data-points-the-idea-behind-statistical-distributions.html#measurements-of-variation"><i class="fa fa-check"></i><b>3.2.2</b> Measurements of variation</a></li>
<li class="chapter" data-level="3.2.3" data-path="how-to-summarize-collection-of-data-points-the-idea-behind-statistical-distributions.html"><a href="how-to-summarize-collection-of-data-points-the-idea-behind-statistical-distributions.html#statistical-distributions"><i class="fa fa-check"></i><b>3.2.3</b> Statistical distributions</a></li>
<li class="chapter" data-level="3.2.4" data-path="how-to-summarize-collection-of-data-points-the-idea-behind-statistical-distributions.html"><a href="how-to-summarize-collection-of-data-points-the-idea-behind-statistical-distributions.html#confidence-intervals"><i class="fa fa-check"></i><b>3.2.4</b> Confidence intervals</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="how-to-test-for-differences-between-samples.html"><a href="how-to-test-for-differences-between-samples.html"><i class="fa fa-check"></i><b>3.3</b> How to test for differences between samples</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="how-to-test-for-differences-between-samples.html"><a href="how-to-test-for-differences-between-samples.html#randomization"><i class="fa fa-check"></i><b>3.3.1</b> Randomization</a></li>
<li class="chapter" data-level="3.3.2" data-path="how-to-test-for-differences-between-samples.html"><a href="how-to-test-for-differences-between-samples.html#t-test"><i class="fa fa-check"></i><b>3.3.2</b> t-test</a></li>
<li class="chapter" data-level="3.3.3" data-path="how-to-test-for-differences-between-samples.html"><a href="how-to-test-for-differences-between-samples.html#multiple-testing"><i class="fa fa-check"></i><b>3.3.3</b> Multiple testing</a></li>
<li class="chapter" data-level="3.3.4" data-path="how-to-test-for-differences-between-samples.html"><a href="how-to-test-for-differences-between-samples.html#moderated-t-tests"><i class="fa fa-check"></i><b>3.3.4</b> Moderated t-tests</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="relationship-between-variables-linear-models-and-correlation.html"><a href="relationship-between-variables-linear-models-and-correlation.html"><i class="fa fa-check"></i><b>3.4</b> Relationship between variables: Linear models and correlation</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="relationship-between-variables-linear-models-and-correlation.html"><a href="relationship-between-variables-linear-models-and-correlation.html#the-cost-or-loss-function-approach"><i class="fa fa-check"></i><b>3.4.1</b> The cost or loss function approach</a></li>
<li class="chapter" data-level="3.4.2" data-path="relationship-between-variables-linear-models-and-correlation.html"><a href="relationship-between-variables-linear-models-and-correlation.html#the-maximum-likelihood-approach"><i class="fa fa-check"></i><b>3.4.2</b> The “maximum likelihood” approach</a></li>
<li class="chapter" data-level="3.4.3" data-path="relationship-between-variables-linear-models-and-correlation.html"><a href="relationship-between-variables-linear-models-and-correlation.html#linear-algebra-and-closed-form-solution-to-linear-regression"><i class="fa fa-check"></i><b>3.4.3</b> Linear algebra and closed-form solution to linear regression</a></li>
<li class="chapter" data-level="3.4.4" data-path="relationship-between-variables-linear-models-and-correlation.html"><a href="relationship-between-variables-linear-models-and-correlation.html#how-to-estimate-the-error-of-the-coefficients"><i class="fa fa-check"></i><b>3.4.4</b> How to estimate the error of the coefficients</a></li>
<li class="chapter" data-level="3.4.5" data-path="relationship-between-variables-linear-models-and-correlation.html"><a href="relationship-between-variables-linear-models-and-correlation.html#accuracy-of-the-model"><i class="fa fa-check"></i><b>3.4.5</b> Accuracy of the model</a></li>
<li class="chapter" data-level="3.4.6" data-path="relationship-between-variables-linear-models-and-correlation.html"><a href="relationship-between-variables-linear-models-and-correlation.html#regression-with-categorical-variables"><i class="fa fa-check"></i><b>3.4.6</b> Regression with categorical variables</a></li>
<li class="chapter" data-level="3.4.7" data-path="relationship-between-variables-linear-models-and-correlation.html"><a href="relationship-between-variables-linear-models-and-correlation.html#regression-pitfalls"><i class="fa fa-check"></i><b>3.4.7</b> Regression pitfalls</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="meeting-videos-2.html"><a href="meeting-videos-2.html"><i class="fa fa-check"></i><b>3.5</b> Meeting Videos</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="meeting-videos-2.html"><a href="meeting-videos-2.html#cohort-1-2"><i class="fa fa-check"></i><b>3.5.1</b> Cohort 1</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="exploratory-data-analysis-with-unsupervised-machine-learning.html"><a href="exploratory-data-analysis-with-unsupervised-machine-learning.html"><i class="fa fa-check"></i><b>4</b> Exploratory Data Analysis with Unsupervised Machine Learning</a>
<ul>
<li class="chapter" data-level="4.1" data-path="very-helpful-youtube-explanations.html"><a href="very-helpful-youtube-explanations.html"><i class="fa fa-check"></i><b>4.1</b> Very helpful youtube explanations</a></li>
<li class="chapter" data-level="4.2" data-path="chapter-4-exercises.html"><a href="chapter-4-exercises.html"><i class="fa fa-check"></i><b>4.2</b> Chapter 4 Exercises</a></li>
<li class="chapter" data-level="4.3" data-path="meeting-videos-3.html"><a href="meeting-videos-3.html"><i class="fa fa-check"></i><b>4.3</b> Meeting Videos</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="meeting-videos-3.html"><a href="meeting-videos-3.html#cohort-1-3"><i class="fa fa-check"></i><b>4.3.1</b> Cohort 1</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="predictive-modeling-with-supervised-machine-learning.html"><a href="predictive-modeling-with-supervised-machine-learning.html"><i class="fa fa-check"></i><b>5</b> Predictive Modeling with Supervised Machine Learning</a>
<ul>
<li class="chapter" data-level="5.1" data-path="and-5.html"><a href="and-5.html"><i class="fa fa-check"></i><b>5.1</b> 5.1 and 5.2</a></li>
<li class="chapter" data-level="5.2" data-path="data-pre-processing.html"><a href="data-pre-processing.html"><i class="fa fa-check"></i><b>5.2</b> 5.4 Data pre-processing</a></li>
<li class="chapter" data-level="5.3" data-path="split-the-data.html"><a href="split-the-data.html"><i class="fa fa-check"></i><b>5.3</b> 5.5 Split the data</a></li>
<li class="chapter" data-level="5.4" data-path="some-unsupervised-learning-on-training-data-to-initially-cluster-data.html"><a href="some-unsupervised-learning-on-training-data-to-initially-cluster-data.html"><i class="fa fa-check"></i><b>5.4</b> 5.6 Some unsupervised learning on training data to initially cluster data</a></li>
<li class="chapter" data-level="5.5" data-path="section.html"><a href="section.html"><i class="fa fa-check"></i><b>5.5</b> 5.7</a></li>
<li class="chapter" data-level="5.6" data-path="section-1.html"><a href="section-1.html"><i class="fa fa-check"></i><b>5.6</b> 5.8</a></li>
<li class="chapter" data-level="5.7" data-path="meeting-videos-4.html"><a href="meeting-videos-4.html"><i class="fa fa-check"></i><b>5.7</b> Meeting Videos</a>
<ul>
<li class="chapter" data-level="5.7.1" data-path="meeting-videos-4.html"><a href="meeting-videos-4.html#cohort-1-4"><i class="fa fa-check"></i><b>5.7.1</b> Cohort 1</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="operations-on-genomic-intervals-and-genome-arithmetic.html"><a href="operations-on-genomic-intervals-and-genome-arithmetic.html"><i class="fa fa-check"></i><b>6</b> Operations on Genomic Intervals and Genome Arithmetic</a>
<ul>
<li class="chapter" data-level="6.1" data-path="slide-1.html"><a href="slide-1.html"><i class="fa fa-check"></i><b>6.1</b> SLIDE 1</a></li>
<li class="chapter" data-level="6.2" data-path="meeting-videos-5.html"><a href="meeting-videos-5.html"><i class="fa fa-check"></i><b>6.2</b> Meeting Videos</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="meeting-videos-5.html"><a href="meeting-videos-5.html#cohort-1-5"><i class="fa fa-check"></i><b>6.2.1</b> Cohort 1</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="quality-check-processing-and-alignment-of-high-throughput-sequencing-reads.html"><a href="quality-check-processing-and-alignment-of-high-throughput-sequencing-reads.html"><i class="fa fa-check"></i><b>7</b> Quality Check, Processing and Alignment of High-throughput Sequencing Reads</a>
<ul>
<li class="chapter" data-level="7.1" data-path="slide-1-1.html"><a href="slide-1-1.html"><i class="fa fa-check"></i><b>7.1</b> SLIDE 1</a></li>
<li class="chapter" data-level="7.2" data-path="meeting-videos-6.html"><a href="meeting-videos-6.html"><i class="fa fa-check"></i><b>7.2</b> Meeting Videos</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="meeting-videos-6.html"><a href="meeting-videos-6.html#cohort-1-6"><i class="fa fa-check"></i><b>7.2.1</b> Cohort 1</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="rna-seq-analysis.html"><a href="rna-seq-analysis.html"><i class="fa fa-check"></i><b>8</b> RNA-seq Analysis</a>
<ul>
<li class="chapter" data-level="8.1" data-path="slide-1-2.html"><a href="slide-1-2.html"><i class="fa fa-check"></i><b>8.1</b> SLIDE 1</a></li>
<li class="chapter" data-level="8.2" data-path="meeting-videos-7.html"><a href="meeting-videos-7.html"><i class="fa fa-check"></i><b>8.2</b> Meeting Videos</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="meeting-videos-7.html"><a href="meeting-videos-7.html#cohort-1-7"><i class="fa fa-check"></i><b>8.2.1</b> Cohort 1</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="chip-seq-analysis.html"><a href="chip-seq-analysis.html"><i class="fa fa-check"></i><b>9</b> ChIP-seq analysis</a>
<ul>
<li class="chapter" data-level="9.1" data-path="slide-1-3.html"><a href="slide-1-3.html"><i class="fa fa-check"></i><b>9.1</b> SLIDE 1</a></li>
<li class="chapter" data-level="9.2" data-path="meeting-videos-8.html"><a href="meeting-videos-8.html"><i class="fa fa-check"></i><b>9.2</b> Meeting Videos</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="meeting-videos-8.html"><a href="meeting-videos-8.html#cohort-1-8"><i class="fa fa-check"></i><b>9.2.1</b> Cohort 1</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="dna-methylation-analysis-using-bisulfite-sequencing-data.html"><a href="dna-methylation-analysis-using-bisulfite-sequencing-data.html"><i class="fa fa-check"></i><b>10</b> DNA methylation analysis using bisulfite sequencing data</a>
<ul>
<li class="chapter" data-level="10.1" data-path="slide-1-4.html"><a href="slide-1-4.html"><i class="fa fa-check"></i><b>10.1</b> SLIDE 1</a></li>
<li class="chapter" data-level="10.2" data-path="meeting-videos-9.html"><a href="meeting-videos-9.html"><i class="fa fa-check"></i><b>10.2</b> Meeting Videos</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="meeting-videos-9.html"><a href="meeting-videos-9.html#cohort-1-9"><i class="fa fa-check"></i><b>10.2.1</b> Cohort 1</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="multi-omics-analysis.html"><a href="multi-omics-analysis.html"><i class="fa fa-check"></i><b>11</b> Multi-omics Analysis</a>
<ul>
<li class="chapter" data-level="11.1" data-path="slide-1-5.html"><a href="slide-1-5.html"><i class="fa fa-check"></i><b>11.1</b> SLIDE 1</a></li>
<li class="chapter" data-level="11.2" data-path="meeting-videos-10.html"><a href="meeting-videos-10.html"><i class="fa fa-check"></i><b>11.2</b> Meeting Videos</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="meeting-videos-10.html"><a href="meeting-videos-10.html#cohort-1-10"><i class="fa fa-check"></i><b>11.2.1</b> Cohort 1</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Computational Genomics with R Book Club</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="reading-and-writing-data" class="section level2 hasAnchor" number="2.4">
<h2><span class="header-section-number">2.4</span> Reading and writing data<a href="reading-and-writing-data.html#reading-and-writing-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>First download the <code>compGenomRData</code> package from Github if you haven’t done so yet.
We are using this as a companion for the course!</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="reading-and-writing-data.html#cb5-1" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;compgenomr/compGenomRData&quot;</span>)</span></code></pre></div>
<p>Read files in base R</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="reading-and-writing-data.html#cb6-1" aria-hidden="true" tabindex="-1"></a>cpgiFilePath<span class="ot">=</span><span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>,</span>
<span id="cb6-2"><a href="reading-and-writing-data.html#cb6-2" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot;subset.cpgi.hg18.bed&quot;</span>,</span>
<span id="cb6-3"><a href="reading-and-writing-data.html#cb6-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">package=</span><span class="st">&quot;compGenomRData&quot;</span>)</span>
<span id="cb6-4"><a href="reading-and-writing-data.html#cb6-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-5"><a href="reading-and-writing-data.html#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="co"># read CpG island BED file</span></span>
<span id="cb6-6"><a href="reading-and-writing-data.html#cb6-6" aria-hidden="true" tabindex="-1"></a>cpgi.df <span class="ot">&lt;-</span> <span class="fu">read.table</span>(cpgiFilePath, <span class="at">header =</span> <span class="cn">FALSE</span>) </span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="reading-and-writing-data.html#cb7-1" aria-hidden="true" tabindex="-1"></a>enhancerFilePath<span class="ot">=</span><span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>,</span>
<span id="cb7-2"><a href="reading-and-writing-data.html#cb7-2" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot;subset.enhancers.hg18.bed&quot;</span>,</span>
<span id="cb7-3"><a href="reading-and-writing-data.html#cb7-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">package=</span><span class="st">&quot;compGenomRData&quot;</span>)</span>
<span id="cb7-4"><a href="reading-and-writing-data.html#cb7-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-5"><a href="reading-and-writing-data.html#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="co"># read enhancer marker BED file</span></span>
<span id="cb7-6"><a href="reading-and-writing-data.html#cb7-6" aria-hidden="true" tabindex="-1"></a>enh.df <span class="ot">&lt;-</span> <span class="fu">read.table</span>(enhancerFilePath, <span class="at">header =</span> <span class="cn">FALSE</span>) </span>
<span id="cb7-7"><a href="reading-and-writing-data.html#cb7-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-8"><a href="reading-and-writing-data.html#cb7-8" aria-hidden="true" tabindex="-1"></a><span class="co"># check first lines to see how the data looks like</span></span>
<span id="cb7-9"><a href="reading-and-writing-data.html#cb7-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(enh.df)</span></code></pre></div>
<pre><code>##      V1     V2     V3 V4   V5 V6    V7      V8 V9
## 1 chr20 266275 267925  . 1000  .  9.11 13.1693 -1
## 2 chr20 287400 294500  . 1000  . 10.53 13.0231 -1
## 3 chr20 300500 302500  . 1000  .  9.10 13.3935 -1
## 4 chr20 330400 331800  . 1000  .  6.39 13.5105 -1
## 5 chr20 341425 343400  . 1000  .  6.20 12.9852 -1
## 6 chr20 437975 439900  . 1000  .  6.31 13.5184 -1</code></pre>
<p>the key here is</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="reading-and-writing-data.html#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">read.table</span>()</span></code></pre></div>
<p>Reading large tables with <code>readr</code>.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="reading-and-writing-data.html#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr)</span>
<span id="cb10-2"><a href="reading-and-writing-data.html#cb10-2" aria-hidden="true" tabindex="-1"></a>df.f2<span class="ot">=</span><span class="fu">read_table</span>(enhancerFilePath, <span class="at">col_names =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>## 
## ── Column specification ────────────────────────────────────────────────────────
## cols(
##   X1 = col_character(),
##   X2 = col_double(),
##   X3 = col_double(),
##   X4 = col_character(),
##   X5 = col_double(),
##   X6 = col_character(),
##   X7 = col_double(),
##   X8 = col_double(),
##   X9 = col_double()
## )</code></pre>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="reading-and-writing-data.html#cb12-1" aria-hidden="true" tabindex="-1"></a>df.f2</span></code></pre></div>
<pre><code>## # A tibble: 50,416 × 9
##    X1        X2     X3 X4       X5 X6       X7    X8    X9
##    &lt;chr&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt; &lt;dbl&gt; &lt;chr&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;
##  1 chr20 266275 267925 .      1000 .      9.11  13.2    -1
##  2 chr20 287400 294500 .      1000 .     10.5   13.0    -1
##  3 chr20 300500 302500 .      1000 .      9.1   13.4    -1
##  4 chr20 330400 331800 .      1000 .      6.39  13.5    -1
##  5 chr20 341425 343400 .      1000 .      6.2   13.0    -1
##  6 chr20 437975 439900 .      1000 .      6.31  13.5    -1
##  7 chr20 516650 518525 .      1000 .     12.5   13.7    -1
##  8 chr20 519100 521475 .      1000 .      7.1   13.1    -1
##  9 chr20 543800 545775 .      1000 .      9.52  13.0    -1
## 10 chr20 573550 574975 .      1000 .      7.71  13.7    -1
## # … with 50,406 more rows</code></pre>
<p>Readr is part of the tidyverse (tidyverse is life).</p>
<p>Writing data w/ <code>write.table</code></p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="reading-and-writing-data.html#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(enh.df,<span class="at">file=</span><span class="st">&quot;enh.txt&quot;</span>,<span class="at">quote=</span><span class="cn">FALSE</span>,</span>
<span id="cb14-2"><a href="reading-and-writing-data.html#cb14-2" aria-hidden="true" tabindex="-1"></a>            <span class="at">row.names=</span><span class="cn">FALSE</span>,<span class="at">col.names=</span><span class="cn">FALSE</span>,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span></code></pre></div>
<ul>
<li>The type of file it is save as depends on the sep (^ saved a tab separated file or .tsv)</li>
</ul>
<p>Saving/loading R objects directly into/from a file</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="reading-and-writing-data.html#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># save() saves many objects at once, regardless of class</span></span>
<span id="cb15-2"><a href="reading-and-writing-data.html#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(cpgi.df,enh.df,<span class="at">file=</span><span class="st">&quot;mydata.RData&quot;</span>)</span>
<span id="cb15-3"><a href="reading-and-writing-data.html#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;mydata.RData&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="reading-and-writing-data.html#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># saveRDS() can save one object at a type</span></span>
<span id="cb16-2"><a href="reading-and-writing-data.html#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(cpgi.df,<span class="at">file=</span><span class="st">&quot;cpgi.rds&quot;</span>)</span>
<span id="cb16-3"><a href="reading-and-writing-data.html#cb16-3" aria-hidden="true" tabindex="-1"></a>x<span class="ot">=</span><span class="fu">readRDS</span>(<span class="st">&quot;cpgi.rds&quot;</span>)</span>
<span id="cb16-4"><a href="reading-and-writing-data.html#cb16-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(x)</span>
<span id="cb16-5"><a href="reading-and-writing-data.html#cb16-5" aria-hidden="true" tabindex="-1"></a><span class="co"># when using saveRDS() must assign output of readRDS() to new variable in session</span></span></code></pre></div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="data-stuctures-and-data-types.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="plots-base-r-vs-ggplot.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/r4ds/bookclub-comp_genom/edit/main/02.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
